### PR TITLE
PageTransport and actionElement

### PR DESCRIPTION
**Pull request by jpgonzales27**

### CODE REVIEW

File: src/main/java/core/selenium/WebDriverConfig.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * This software is the confidential and proprietary information of Fundacion Jala
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ */
+
+package core.selenium;
+
+import core.utils.PropertiesReader;
+
```
Comment (GustavoZHA) [inline src/main/java/core/selenium/WebDriverConfig.java]: delete blank lines

File: src/main/java/core/selenium/WebDriverConfig.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    /**
+     * Constructor of page WebDriverManager.
+     */
+    private WebDriverConfig() {
+        initialize();
+    }
+
+    /**
+     * Constructo of WebDriverConfig.
```
Comment (GustavoZHA) [inline src/main/java/core/selenium/WebDriverConfig.java]: I would be better instances or build or some thing like that.Use third person in verbs

File: config.properties [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#ENVIRONMET CONFIGURATION
+baseUrl=https://jpcorp-dev-ed.my.salesforce.com
```
Comment (GustavoZHA) [inline config.properties:2]: They are constans, they are in uppercase

File: src/main/java/core/selenium/WebDriverConfig.java (hunk 1 above)
Comment (GustavoZHA) [reply src/main/java/core/selenium/WebDriverConfig.java]: Check all iimport rest of classes

File: src/main/java/core/selenium/WebDriverManager.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    /**
+     * Initializes According the config file.
+     */
+    private void initialize() {
+        io.github.bonigarcia.wdm.WebDriverManager.chromedriver().setup();
+        chromeOptions = new ChromeOptions();
+        chromeOptions.addArguments("--disable-notifications");
+        this.driver = new ChromeDriver(chromeOptions);
```
Comment (GustavoZHA) [inline src/main/java/core/selenium/WebDriverManager.java]: I recommend manager browser in other class or set in a method

File: src/main/java/core/selenium/WebDriverManager.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ * license agreement you entered into with Fundacion Jala
+ */
+
+package core.selenium;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
```
Comment (JessickaMoya) [inline src/main/java/core/selenium/WebDriverManager.java]: Delete blank line

File: src/main/java/core/utils/PropertiesReader.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+     *
+     * @param filePath Json file path.
+     * @return a property entity.
+     */
+    public static Properties getProperties(final String filePath) {
+        Properties properties = new Properties();
+        try {
+            InputStream reading = new FileInputStream(filePath);
+            properties.load(reading);
+        } catch (IOException | NullPointerException e) {
```
Comment (JessickaMoya) [inline src/main/java/core/utils/PropertiesReader.java]: You should change the name of the variable "e" for a more specific name

File: src/main/java/core/utils/StringToDate.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                calendar.add(Calendar.DAY_OF_MONTH, +1);
+                return calendar.getTime();
+            }
+            String[] actions = cutString(valor);
+            if (actions[actions.length - 1].contains("ago")) {
+                return calculateDate(Integer.parseInt(actions[0]) * -1, actions[1], calendar);
+            } else {
+                return calculateDate(Integer.parseInt(actions[0]), actions[1], calendar);
+            }
+        } catch (RuntimeException | ParseException e) {
```
Comment (JessickaMoya) [inline src/main/java/core/utils/StringToDate.java]: You should change the name of the variable "e" for a more specific name

File: src/main/java/salesforce/ui/PageTransporter.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+import core.selenium.WebDriverManager;
+import salesforce.config.EnvConfig;
+import salesforce.ui.pages.accounts.AccountsPage;
+import salesforce.ui.pages.contacts.ContactsPage;
+
+public class PageTransporter {
+
+    /**
+     * .
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/PageTransporter.java]: I don't think it is necessary to have javadoc here

File: src/main/java/salesforce/ui/PageTransporter.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        WebDriverManager.getInstance().getDriver().navigate().to(url);
+    }
+
+    /**
+     * navigates to Accounts page.
+     *
+     * @return Accounts' instance.
+     */
+    public AccountsPage navigateToAccountsPage() {
+        goToURL(baseUrl.concat("/lightning/o/Account/list?filterName=Recent"));
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/PageTransporter.java]: You should obtain the  Account URL  from your properties file, as you do for your baseUrl

File: src/main/java/salesforce/ui/PageTransporter.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    /**
+     * navigates to Contacts page.
+     *
+     * @return Contacts' instance.
+     */
+    public ContactsPage navigateToContactsPage() {
+        System.out.println(baseUrl + "/lightning/o/Account/list?filterName=Recent");
+        goToURL(baseUrl.concat("/lightning/o/Contact/list?filterName=Recent"));
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/PageTransporter.java]: You should obtain the ContactURL from your properties file, as you do for your baseUrl

File: src/main/java/salesforce/ui/PageTransporter.java [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    /**
+     * navigates to Accounts page.
+     *
+     * @return Accounts' instance.
+     */
+    public AccountsPage navigateToAccountsPage() {
+        goToURL(baseUrl.concat("/lightning/o/Account/list?filterName=Recent"));
+        System.out.println(baseUrl.concat("/lightning/o/Account/list?filterName=Recent"));
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/PageTransporter.java]: Delete print line

File: src/main/java/salesforce/ui/PageTransporter.java [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+        return new AccountsPage();
+    }
+
+    /**
+     * navigates to Contacts page.
+     *
+     * @return Contacts' instance.
+     */
+    public ContactsPage navigateToContactsPage() {
+        System.out.println(baseUrl + "/lightning/o/Account/list?filterName=Recent");
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/PageTransporter.java]: Delete print line

File: src/main/java/salesforce/ui/pages/contacts/ContactsFormPage.java [hunk 13]
```diff
@@ -0,0 +1,407 @@
...
+    @FindBy(name = "firstName")
+    private WebElement firstNameTxtBox;
+
+    @FindBy(name = "salutation")
+    private WebElement salutationComboBox;
+
+    @FindBy(xpath = "//input[@name='lastName']")
+    private WebElement lastNameTxtBox;
+
+//    @FindBy(xpath = "//input[@name='lastName']")
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/contacts/ContactsFormPage.java]: Delete commented lines

File: src/main/java/salesforce/ui/pages/contacts/ContactsFormPage.java [hunk 14]
```diff
@@ -0,0 +1,407 @@
...
+    /**
+     * Sets the lastName field.
+     *
+     * @param lastName lastName's value.
+     */
+    public void setLastNameTxtBox(final String lastName) {
+        webElementAction.setInputField(lastNameTxtBox, lastName);
+    }
+
+//    /**
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/contacts/ContactsFormPage.java]: Delete commented lines

File: src/test/java/accounts/CreateAccountTest.java [hunk 15]
```diff
@@ -1,8 +1,51 @@
...
+import salesforce.ui.pages.accounts.NewAccountPopup;
 
 public class CreateAccountTest extends BaseTest {
-    HomePage homePage = loginPage.loginSuccessful(configProperties.getUsername(),configProperties.getPassword());
+
+    private LoginPage loginPage;
+    private HomePage homePage;
+    private AccountsPage accountsPage;
+    private AccountPage accountPage;
+
```
Comment (JessickaMoya) [inline src/test/java/accounts/CreateAccountTest.java]: Dele blank lines

File: src/main/java/core/selenium/WebDriverManager.java [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+            instance = new WebDriverManager();
+        }
+        return instance;
+    }
+
+    /**
+     * Initializes According the config file.
+     */
+    private void initialize() {
+        io.github.bonigarcia.wdm.WebDriverManager.chromedriver().setup();
```
Comment (Raymundogs) [inline src/main/java/core/selenium/WebDriverManager.java]: This solution only works for chromedriver. You should refactor it to allow more types of drivers.

File: src/main/java/core/utils/DateToString.java [hunk 17]
```diff
@@ -0,0 +1,27 @@
+package core.utils;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DateToString {
+
+    /**
+     * .
```
Comment (Raymundogs) [inline src/main/java/core/utils/DateToString.java]: you should not have empty javadocs.

File: src/main/java/core/utils/DateToString.java [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+        SimpleDateFormat df = new SimpleDateFormat("MMddyyyyhhmmss");
+        return df.format(fecha);
+    }
+
+    /**
+     * .
+     * @return .
+     */
+    public static String currentDateToString() {
+        Date fecha = new Date();
```
Comment (Raymundogs) [inline src/main/java/core/utils/DateToString.java]: variables names should be in english.

File: src/main/java/core/utils/StringToDate.java [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+    SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+
+    /**
+     * Transforms a string in a date type.
+     *
+     * @param value string with dates information.
+     * @return a Date value.
+     */
+    public Date convertirString(final String value) {
+        String valor = value.toLowerCase();
```
Comment (Raymundogs) [inline src/main/java/core/utils/StringToDate.java:27]: variable's names should be in english.

File: src/main/java/salesforce/ui/pages/accounts/AccountPage.java [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+
+package salesforce.ui.pages.accounts;
+
+
+import salesforce.ui.pages.BasePage;
+
+public class AccountPage extends BasePage {
+
+    @Override
+    protected void waitUntilPageObjectIsLoaded() {
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/accounts/AccountPage.java:26]: You should not have empty methods.

File: src/main/java/salesforce/ui/pages/contacts/ContactsFormPage.java [hunk 21]
```diff
@@ -0,0 +1,242 @@
...
+    private WebElement mailingCountryTxtBox;
+
+    @FindBy(xpath = "(//textarea[@name='street'])[2]")
+    private WebElement otherStreetTxtBox;
+
+    @FindBy(xpath = "(//input[@name='postalCode'])[2]")
+    private WebElement otherPostalCodeTxtBox;
+
+    @FindBy(xpath = "(//input[@name='city'])[2]")
+    private WebElement otherCityTxtBox;
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/contacts/ContactsFormPage.java]: Some xpath are repeated, maybe you could refactor them. Also you should avoid using indexes on the locators.

File: src/main/java/salesforce/ui/pages/accounts/NewAccountPopup.java [hunk 22]
```diff
@@ -0,0 +1,86 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import salesforce.ui.pages.BasePage;
+
+import java.util.HashMap;
+
+public class NewAccountPopup extends BasePage {
+
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/accounts/NewAccountPopup.java:21]: I think you are not using this web element anymore. You should delete it. The same goes for faxTxtBox

File: src/main/java/salesforce/ui/pages/contacts/ContactPage.java [hunk 23]
```diff
@@ -0,0 +1,114 @@
...
+ */
+
+package salesforce.ui.pages.contacts;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import salesforce.ui.pages.BasePage;
+
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/contacts/ContactPage.java]: Delete blank line

File: src/main/java/salesforce/ui/pages/contacts/ContactPage.java [hunk 24]
```diff
@@ -0,0 +1,114 @@
...
+public class ContactPage extends BasePage {
+
+    @FindBy(css = ".slds-theme--success")
+    private WebElement alertSuccess;
+
+    @FindBy(xpath = "//a[@id='detailTab__item']")
+    private WebElement detailsTab;
+
+    //    @FindBy(xpath = "//span[normalize-space()='Dr. Contact Test']")
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/contacts/ContactPage.java]: Delete commented line

File: src/main/java/salesforce/ui/pages/contacts/ContactPage.java [hunk 25]
```diff
@@ -0,0 +1,114 @@
...
+     * Gets the alert text.
+     *
+     * @return the text value.
+     */
+    public String getTextAlertSuccess() {
+        return alertSuccess.getText();
+    }
+
+    /**
+     * .
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/contacts/ContactPage.java]: Complete your javadocs

File: src/test/java/contacts/CreateContactTest.java [hunk 26]
```diff
@@ -0,0 +1,105 @@
...
+        softAssert.assertEquals(contactLastName, contactPage.getPrincipalName());
+        contactsPage = pageTransporter.navigateToContactsPage();
+        softAssert.assertEquals(contactLastName,contactsPage.getNamesText(contactLastName));
+        softAssert.assertAll();
+    }
+
+
+    @Test
+    public void createContactWithAllFields() {
+        String salutation = "Dr.";
```
Comment (JessickaMoya) [inline src/test/java/contacts/CreateContactTest.java]: Maybe you can declare this variables outside the test, so you can reuse them in other tests

File: src/main/java/core/selenium/WebDriverManager.java (hunk 4 above)
Comment (jpgonzales27) [reply src/main/java/core/selenium/WebDriverManager.java]: This class is the manager browser in this class sets all configurations

**Review by JessickaMoya — APPROVED** [review]

File: src/main/java/salesforce/ui/pages/contacts/ContactPage.java [hunk 27]
```diff
@@ -0,0 +1,111 @@
...
+     *
+     * @param fieldName .
+     * @return a String with the entity or company name text.
+     */
+    public String getDateText(final String fieldName) {
+        return webElementAction.getElementText(driver.findElement(By.xpath(String.format(SPAN_DATE, fieldName))));
+    }
+
+    /**
+     * .
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/contacts/ContactPage.java]: This three methods are missing a description.

**Review by Raymundogs — APPROVED** [review]

**Review by GustavoZHA — APPROVED** [review]

**Pull request merged**